main.go

```diff
@@ -2,51 +2,67 @@ package main
 
 import (
 	"bufio"
+	"bytes"
+	"encoding/gob"
+	"errors"
 	"fmt"
 	"io/ioutil"
-	"log"
 	"net/http"
 	"os"
+	"os/user"
+	"path/filepath"
 	"runtime"
 	"strings"
+	"time"
 
 	"github.com/fatih/color"
+	"github.com/peterbourgon/diskv"
 )
 
 const pageURLPattern = "https://raw.githubusercontent.com/tldr-pages/tldr/master/pages/%s/%s.md"
 
 func main() {
 	if len(os.Args) != 2 {
-		log.Fatal("Usage: tldr <command name>\n")
+		fmt.Fprint(os.Stderr, "What tlrd page do you want?\n")
+		os.Exit(-1)
 	}
 
-	command := os.Args[1]
-	page, err := fetchPage(getPlatform(), command)
+	var content string
+	var err error
 
-	if err != nil {
-		fmt.Println(err)
+	page := os.Args[1]
+
+	if content, err = fetchPage(getPlatform(), page); err != nil {
+		fmt.Fprintf(os.Stderr, "Page not found: %s\n", page)
 		os.Exit(-1)
 	}
 
-	// Command found, render and print to stdout.
-	fmt.Print(renderPage(page))
+	// Render and print.
+	fmt.Print(renderPage(content))
 }
 
-func fetchPage(platform, command string) (string, error) {
-	pages := make(chan *string)
+func fetchPage(platform, page string) (string, error) {
+	responses := make(chan *string)
 
-	// Fetch the command from platform and common directories in parallel.
+	// Fetch the page from platform and common directories in parallel.
 	// Then discard the one that fails.
-	go fetch(fmt.Sprintf(pageURLPattern, "common", command), pages)
-	go fetch(fmt.Sprintf(pageURLPattern, platform, command), pages)
+	go fetch(fmt.Sprintf(pageURLPattern, "common", page), responses)
+	go fetch(fmt.Sprintf(pageURLPattern, platform, page), responses)
+
+	cache := newCache()
```

Comment:
I can't think of a recommendation, just that when I read newCache() followed by cache.read() my mind goes "there won't be anything on that new cache". Perhaps a rename, but really not sure what's best